### PR TITLE
[BUGFIX] Fix missing tooltip on updated variables

### PR DESCRIPTION
**Pull request by AntoineThebaud**
# Description

Fixes https://github.com/perses/perses/issues/1653.
Empty strings for desc attributes are no longer kept when applying/saving.

# Screenshots

image

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the
  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [ ] Visual tests are stable and unlikely to be flaky.
  See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests)
  and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues
  include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

File: ui/plugin-system/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx [hunk 1]
```diff
@@ -83,7 +83,14 @@ export function VariableEditorForm(props: VariableEditorFormProps) {
   });
 
   const processForm: SubmitHandler<VariableEditorValidationType> = () => {
-    onSave(getVariableDefinitionFromState(state));
+    // reset display attributes to undefined when empty, because we don't want to save empty strings
+    onSave(
+      getVariableDefinitionFromState({
+        ...state,
+        title: state.title == '' ? undefined : state.title,
```
Comment (Gladorme) [inline ui/plugin-system/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx:90]:
```suggestion
        title: state.title === '' ? undefined : state.title,
```

File: ui/plugin-system/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx [hunk 2]
```diff
@@ -83,7 +83,14 @@ export function VariableEditorForm(props: VariableEditorFormProps) {
...
+        description: state.description == '' ? undefined : state.description,
```
Comment (Gladorme) [inline ui/plugin-system/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx:91]:
```suggestion
        description: state.description === '' ? undefined : state.description,
```

File: ui/plugin-system/src/components/DatasourceEditorForm/DatasourceEditorForm.tsx [hunk 3]
```diff
@@ -74,16 +74,13 @@ export function DatasourceEditorForm(props: DatasourceEditorFormProps) {
...
-      name: name ? name : undefined,
-    };
-
+    // reset display attributes to undefined when empty, because we don't want to save empty strings
     onSave(state.name, {
       ...state.spec,
-      display: finalDisplay,
+      display: {
+        name: state.spec.display?.name == '' ? undefined : state.spec.display?.name,
+        description: state.spec.display?.description == '' ? undefined : state.spec.display?.description,
```
Comment (Gladorme) [inline ui/plugin-system/src/components/DatasourceEditorForm/DatasourceEditorForm.tsx:82]:
```suggestion
        description: state.spec.display?.description === '' ? undefined : state.spec.display?.description,
```

File: ui/plugin-system/src/components/DatasourceEditorForm/DatasourceEditorForm.tsx [hunk 4]
```diff
@@ -74,16 +74,13 @@ export function DatasourceEditorForm(props: DatasourceEditorFormProps) {
...
-      ...state.spec.display,
-      name: name ? name : undefined,
-    };
-
+    // reset display attributes to undefined when empty, because we don't want to save empty strings
     onSave(state.name, {
       ...state.spec,
-      display: finalDisplay,
+      display: {
+        name: state.spec.display?.name == '' ? undefined : state.spec.display?.name,
```
Comment (Gladorme) [inline ui/plugin-system/src/components/DatasourceEditorForm/DatasourceEditorForm.tsx:81]:
```suggestion
        name: state.spec.display?.name === '' ? undefined : state.spec.display?.name,
```

**Review by Gladorme — COMMENTED** [review]
We could also have a check on the variable input, if display name is empty => show "metdata" name 🤔 Idea for a futur PR...

**Comment by AntoineThebaud** [comment]
> We could also have a check on the variable input, if display name is empty => show "metdata" name 🤔 Idea for a futur PR...

It's already what happening, cf above screenshot, or did I misunderstood your point?

**Comment by Gladorme** [comment]
> > We could also have a check on the variable input, if display name is empty => show "metdata" name 🤔 Idea for a futur PR...
> 
> It's already what happening, cf above screenshot, or did I misunderstood your point?

Here we are fixing at form level, but we could fix at input level. If there is display { name: ""} : if the display name is empty use metadata name

**Review by Gladorme — APPROVED** [review]

**Comment by AntoineThebaud** [comment]
> > > We could also have a check on the variable input, if display name is empty => show "metdata" name 🤔 Idea for a futur PR...
> > 
> > 
> > It's already what happening, cf above screenshot, or did I misunderstood your point?
> 
> Here we are fixing at form level, but we could fix at input level. If there is display { name: ""} : if the display name is empty use metadata name

Oh ok got it, indeed.

**Pull request merged**